### PR TITLE
add concurrent lock

### PR DESCRIPTION
**Pull request by jairhenrique**

### CODE REVIEW

**Review by RomuloOliveira — APPROVED** [review]
👍

**Review by ghost — APPROVED** [review]

File: django_toolkit/concurrent/locks.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        self.active = True
+        return self
+
+    def __exit__(self, *args, **kwargs):
+        self.active = False
+
+
+class CacheLock(BaseLock):
+
+    cache = caches['locks']
```
Comment (renanivo) [inline django_toolkit/concurrent/locks.py]: What if we pass the cache name in a parameter? `locks` could be the default value in the constructor.

File: django_toolkit/concurrent/locks.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    def __init__(self, key, expire=0):
+        super(CacheLock, self).__init__()
+        self._key = key
+        self._expire = expire
+
+    def __enter__(self):
+        self.active = self.cache.add(self._key, True, self._expire)
+
+        if not self.active:
+            raise LockActiveError('For key {key}'.format(key=self._key))
```
Comment (renanivo) [inline django_toolkit/concurrent/locks.py]: The user of this context manager would have add a `try... except` block to catch `LockActiveError` and finish the task when it happens, right?

**Review by renanivo — COMMENTED** [review]
It would be good to have docs for this new module.

**Review by carlagouveia-luizalabs — APPROVED** [review]

File: docs/concurrent.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
+# Concurrent
+
+## Locks
+
+The lock context are used to ensure that piece of code run only once.
```
Comment (renanivo) [inline docs/concurrent.md]: What do you think?

``` diff
- are used to ensure that piece of code run only once.
+ is used to indicate that a block of code entered in a [critical section](https://en.wikipedia.org/wiki/Critical_section)
```

File: docs/concurrent.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+def run(*args, **kwargs):
+    with LocalMemoryLock() as lock:
+        assert lock.active is True
+        # do some stuff
+```
+
+### Cache lock
+
+`CacheLock` uses django cache system and `default` cache alias by default.
+If the context of lock is `active`, by default, `LockActiveError` is raised.
```
Comment (renanivo) [inline docs/concurrent.md]: What about?

``` diff
- If the context of lock is `active`, by default, `LockActiveError` is raised.
+ By default, the context manager raises `LockActiveError` when the lock was already active.
```

File: django_toolkit/concurrent/locks.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            raise LockActiveError('Lock is already active')
+
+        self.active = True
+        return self
+
+    def __exit__(self, *args, **kwargs):
+        self.active = False
+
+
+class CacheLock(Lock):
```
Comment (renanivo) [inline django_toolkit/concurrent/locks.py:33]: What about some docstring here? You could include `__init__` parameter descriptions, also.

**Pull request merged**